core/src/main/scala/Config.scala

```diff
@@ -654,29 +638,49 @@ object Config {
 
       val schedConfig = kfg.subconfig("infrastructure.scheduler")
 
-      val schedAndHealthChecker = schedConfig.lookup[String]("scheduler") match {
+      val components = schedConfig.lookup[String]("scheduler") match {
         case Some("nomad") =>
+          val consul = {
+            val a = kfg.require[String]("infrastructure.consul.endpoint")
+            val b = kfg.require[Duration]("infrastructure.consul.timeout")
+            val c = kfg.lookup[String]("infrastructure.consul.acl-token")
+            val d = kfg.lookup[String]("infrastructure.consul.username")
+            val e = kfg.lookup[String]("infrastructure.consul.password")
+            val client = http4sClient(b, 20)
+            val http4sConsul = (d,e) match {
+              case (None,None) => client.map(consulClient => Http4sConsul.client(Infrastructure.Consul(new URI(a), b, c, None), consulClient))
+              case (Some(u),Some(pw)) =>
+                client.map(consulClient => Http4sConsul.client(Infrastructure.Consul(new URI(a), b, c,
+                  Some(Infrastructure.Credentials(u,pw))), consulClient))
+              case _ =>
+                log.error("If you configure the datacenter to have a consul username, or consul password, it must have both.")
+                client.map(consulClient => Http4sConsul.client(Infrastructure.Consul(new URI(a), b, c, None), consulClient))
+            }
+            http4sConsul.map(consulClient => PrometheusConsul(a, consulClient))
+          }
+
           for {
             consulClient  <- consul
             sched         <- readNomadScheduler(schedConfig.subconfig("nomad"))
             healthChecker = health.Http4sConsulHealthClient(consulClient)
-          } yield (sched, healthChecker)
+          } yield (sched, healthChecker, consulClient)
+
         case Some("kubernetes") =>
           readKubernetesClient(schedConfig.subconfig("kubernetes")).map { client =>
-            (new scheduler.KubernetesHttp(client), health.KubernetesHealthClient(client))
+            (new scheduler.KubernetesHttp(client), health.KubernetesHealthClient(client), StubbedConsulClient)
```

Comment:
`StubbedConsulClient`? Would be good to understand the rationale here.

Reply:
I confess this was selfishly motivated - we are not using Consul with Kubernetes not are we (or you afaik) planning on it, but in retrospect it is totally possible (I think [Stripe](https://stripe.com/blog/operating-kubernetes) maybe does this?).

Anyways assuming that is the case, there is no sane values to put for the Consul portion of the config (which we need to make more generic BTW) and so when Nelson goes to refresh routing tables you get this nice stack trace in the logs that is bogus.

Unsure what to do here:

- The Correct Way™ would be to give an option in the config to signal what backend to use for the routing layer, similar to what we do for the scheduler
- Try to make a dummy call to Consul and if that fails fallback to the stubbed client

This of course is a stopgap while we're in the "Kubernetes without Consul currently has no routing support from Nelson" stage. Curious to hear your thoughts